### PR TITLE
Cleanup small spots in CCR

### PR DESCRIPTION
**Pull request by original-brownbear**
Removing some dead code and other obvious small things in CCR. Also, cleaning up cluster state request instances with needlessly large scope to make the code more readable.

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/bulk/TransportBulkShardOperationsAction.java [hunk 1]
```diff
@@ -180,12 +180,11 @@ public static WritePrimaryResult<BulkShardOperationsRequest, BulkShardOperations
                 appliedOperations.add(targetOp);
                 location = locationToSync(location, result.getTranslogLocation());
             } else {
-                if (result.getFailure() instanceof AlreadyProcessedFollowingEngineException) {
+                if (result.getFailure()instanceof final AlreadyProcessedFollowingEngineException failure) {
```
Comment (kingherc) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/bulk/TransportBulkShardOperationsAction.java:183]: Space before `instanceof` please

**Review by kingherc — APPROVED** [review]
Looks like some nice refactoring to me.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardChangesAction.java [hunk 2]
```diff
@@ -214,56 +214,54 @@ public String toString() {
 
     public static final class Response extends ActionResponse {
 
-        private long mappingVersion;
+        private final long mappingVersion;
```
Comment (idegtiarenko) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardChangesAction.java:217]: That is one very unusual field ordering :)
Could this one also be converted to a record?

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/bulk/TransportBulkShardOperationsAction.java (hunk 1 above)
Comment (idegtiarenko) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/bulk/TransportBulkShardOperationsAction.java:183]: I wonder if this is how it is enforced by the spotless?

**Review by idegtiarenko — APPROVED** [review]

**Comment by original-brownbear** [comment]
Thanks both!

**Pull request merged**